Userland/Libraries/LibWeb/Painting/FilterPainting.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2022, MacDue <macdue@dueutil.tech>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibGfx/Filters/BrightnessFilter.h>
+#include <LibGfx/Filters/ContrastFilter.h>
+#include <LibGfx/Filters/GrayscaleFilter.h>
+#include <LibGfx/Filters/InvertFilter.h>
+#include <LibGfx/Filters/OpacityFilter.h>
+#include <LibGfx/Filters/SepiaFilter.h>
+#include <LibGfx/Filters/StackBlurFilter.h>
+#include <LibWeb/Layout/Node.h>
+#include <LibWeb/Painting/BorderRadiusCornerClipper.h>
+#include <LibWeb/Painting/FilterPainting.h>
+
+namespace Web::Painting {
+
+void apply_filter_list(Gfx::Bitmap& target_bitmap, Layout::Node const& node, Span<CSS::FilterFunction const> filter_list)
+{
+    for (auto& filter_function : filter_list) {
+        filter_function.visit(
+            [&](CSS::Filter::Blur const& blur) {
+                // Applies a Gaussian blur to the input image.
+                // The passed parameter defines the value of the standard deviation to the Gaussian function.
+                Gfx::StackBlurFilter filter { target_bitmap };
+                filter.process_rgba(blur.resolved_radius(node), Color::Transparent);
+            },
+            [&](CSS::Filter::Color const& color) {
+                auto amount = color.resolved_amount();
+                auto amount_clammped = clamp(amount, 0.0f, 1.0f);
+                auto apply_color_filter = [&](Gfx::ColorFilter const& filter) {
+                    const_cast<Gfx::ColorFilter&>(filter).apply(target_bitmap, target_bitmap.rect(), target_bitmap, target_bitmap.rect());
```

Comment:
Hmmm. Is there some reason `Filter::apply()` can't be `const`? It feels like it shouldn't have any state that's modified. (That's a tangent though!)

Reply:
I'm pretty sure it could (seems the case for the all the existing filters), but I thought I'd just leave the API be for now